pdns/dnsdistdist/dnsdist-tcp.cc

```diff
@@ -449,14 +449,15 @@ void IncomingTCPConnectionState::updateIOForAsync(std::shared_ptr<IncomingTCPCon
   conn->d_ioState->update(IOState::Done, handleIOCallback, conn);
 }
 
-void IncomingTCPConnectionState::updateIO(std::shared_ptr<IncomingTCPConnectionState>& state, IOState newState, const struct timeval& now)
+void IncomingTCPConnectionState::updateIO(IOState newState, const struct timeval& now)
 {
+  auto sharedPtrToConn = shared_from_this();
   if (newState == IOState::Async) {
-    updateIOForAsync(state);
+    updateIOForAsync(sharedPtrToConn);
     return;
   }
 
-  state->d_ioState->update(newState, handleIOCallback, state, newState == IOState::NeedWrite ? state->getClientWriteTTD(now) : state->getClientReadTTD(now));
+  d_ioState->update(newState, handleIOCallback, sharedPtrToConn, newState == IOState::NeedWrite ? getClientWriteTTD(now) : getClientReadTTD(now));
```

Comment:
or `sharedPtrToConn->d_ioState->update(...)` ?

Reply:
Agreed, it's just making explicit we're calling a method on an object and also passing it as an arg. I'm fine with leaving it as-is.

Reply:
Right, I see your point. I still prefer it the way it is now because it prevents having to wonder why we are using the shared pointer to call this function (why we have to pass it to the function is more obvious).